CMakeLists.txt

```diff
@@ -1524,11 +1524,14 @@ endif()
 
 if(CONFIG_BUILD_OUTPUT_ADJUST_LMA)
   math(EXPR adjustment "${CONFIG_BUILD_OUTPUT_ADJUST_LMA}" OUTPUT_FORMAT DECIMAL)
+  set(args_adjustment ${CONFIG_BUILD_OUTPUT_ADJUST_LMA_SECTIONS})
+  list(TRANSFORM args_adjustment PREPEND $<TARGET_PROPERTY:bintools,elfconvert_flag_lma_adjust>)
+  list(TRANSFORM args_adjustment APPEND +${adjustment})
```

Comment:
I think the real blocker here, now that I realize it, is the `lma_adjust` flag being repeated.

For GNU, this translates to:
```
objcopy --change-section-lma *+<value> --change-section-lma !bss+<value> ...
```
But if a CMake wrapper were involved, then this wouldn't work at all:
```
cmake -P elfconvert.cmake -DLMA_ADJUST=*+<value> -DLMA_ADJUST=!bss+<value> ...
```
One solution could be to port existing CMake wrappers to Python, and have `argparse` take care of "append" style arguments, but that's probably overkill.

Another solution could be to have separate flags for `lma_adjust` (number) and `lma_adjust_sections` (list). This would require translation on the part of a CMake wrapper for objcopy, which might be odd as well.

As it stands, I'm not sure how the current toolchain abstraction scheme should handle this feature.

Reply:
> But if a CMake wrapper were involved, then this wouldn't work at all:

you wouldn't believe the kind of tricks one can do in CMake 😉 
```
$ cmake -DLMA_ADJUST="*+1024" -DLMA_ADJUST="!bss+2048" -P elfconvert.cmake 
Elfconvert lma adjustment: -DLMA_ADJUST=*+1024
Elfconvert lma adjustment: -DLMA_ADJUST=!bss+2048
```
Code:
```
cmake_minimum_required(VERSION 3.20)                                                                                                                                 

foreach(i RANGE ${CMAKE_ARGC})                                                                                                                                       
  if(CMAKE_ARGV${i} MATCHES "^-DLMA_ADJUST(.*)")                                                                                                                     
    message("Elfconvert lma adjustment: ${CMAKE_ARGV${i}}")                                                                                                          
  endif()                                                                                                                                                            
endforeach()
```

but as this requires special CMake skills, then it's harder for others to implement such code.

Note, it's generally required to have all `-D`s before `-P <script>` when invoking CMake in script mode.

> As it stands, I'm not sure how the current toolchain abstraction scheme should handle this feature.

yep, regardless of the path, it's gonna add even extra complexity on top of the complexity we already have.